File: TP-OTI-INIT/money_js_sinon/src/money.js
var money= (function(){
	function money(v,curr) {
		if(v < 0){
			throw new moneyInfZero();
		}else if( v === null || curr=== null){
			throw new MoneyNullValueOrNullCurrency();
		}else if( curr != undefined && curr.length > 3){
			throw new CurrencySupTrois();
		}else{
			this.v=v;
			this.curr=curr;
		}

		this._equals = function(otherM){
			return (otherM.getValue()===this.getValue() && otherM.getCurrency().toUpperCase()===this.getCurrency().toUpperCase()) ;
		}
	}

	money.prototype.getCurrency=function () {
		return this.curr.toUpperCase();
	}
	money.prototype.getValue=function () {
		return this.v;
	}
	money.prototype.equals=function (otherM) {
		return (otherM.getValue()==this.getValue() && otherM.getCurrency()==this.getCurrency()) ;
	}
	money.prototype.toString=function toString() {
		return this.getValue()+" ("+this.getCurrency()+")" ;
	}


return money;
})();

File: RENDU_OTI/moneySrc/src/main/java/Money.java


/**
 * @author afrass
 */
public class Money {
    private int value;
    private String currency;


    Money(int value, String currency) {
        if (value < 0)
            throw new InferieurZeroException();
        else
            this.value = value;
        this.currency = currency;
    }

    public int getValue()
    {
        return this.value;
    }

    public String getCurrency()
    {
        return this.currency;
    }

    @Override
    public String toString() {
        return this.getValue()+" ("+this.getCurrency()+")";
    }

    @Override
    public int hashCode() {
        /* ... */
        return 1;
    }

    public boolean _equals(Object o){

        if(((Money)o).getValue() == this.getValue() && ((Money)o).getCurrency().equals(this.getCurrency()))
            return true;
        return false;
    }
    @Override
    public boolean equals(Object o) {
        return _equals(o);
    }


}